### PR TITLE
Adds a new member routine allocate_diag_field_output_buffers() to fmsDiagObject_type

### PR DESCRIPTION
**Pull request by ganganoaa**
**Description**
Update relates to a new addition of member routine, _allocate_diag_field_output_buffer_, to the type _fmsDiagObject_type_ in the _fms_diag_object_mod_.  The routine _fms_diag_output_buffer_create_container_ in the module, _fms_diag_output_buffer_mod_, is declared _public_ for visibility issue. The dependency of module, _fms_diag_output_buffer_mod_, is inserted in _/diag_manager/Makefile.am_. The _use_ statement for the _fms_diag_yaml_mod_ is also updated to include the _DiagYamlFilesVar_type_.

Fixes # (issue)
CI

**How Has This Been Tested?**
Please describe the tests that you ran to verify your changes. Please also note
any relevant details for your test configuration (e.g. compiler, OS).  Include
enough information so someone can reproduce your tests.

**Checklist:**
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] Any dependent changes have been merged and published in downstream modules
- [x] New check tests, if applicable, are included
- [x] `make distcheck` passes

### CODE REVIEW

File: diag_manager/fms_diag_object.F90 [hunk 1]
```diff
@@ -849,4 +850,85 @@ subroutine dump_diag_obj( filename )
...
+  ! Determine dimensions of the field
+  ndims = 0
+  if (this%FMS_diag_fields(field_id)%has_axis_ids()) then
+    axis_ids => this%FMS_diag_fields(field_id)%get_axis_id() !< Get ids of axes of the variable
+    ndims = size(axis_ids) !< Dimensions of the field
+  endif
+
+  ! Loop over a number of fields/buffers where this variable occurs
+  do i = 1, size(this%FMS_diag_fields(field_id)%buffer_ids)
+    buffer_id = this%FMS_diag_fields(field_id)%buffer_ids(i)
```
Comment (uramirez8707) [inline diag_manager/fms_diag_object.F90:879]: Where are `buffer_ids` going to be set?

File: diag_manager/fms_diag_object.F90 [hunk 2]
```diff
@@ -849,4 +850,85 @@ subroutine dump_diag_obj( filename )
...
+    if (num_diurnal_samples .ne. 0) then
+      allocate(axes_length(ndims + 1)) !< Include extra length for the diurnal axis
+      do j = 1, ndims
+        axes_length(j) = this%fms_get_axis_length(axis_ids(j))
+      enddo
+      axes_length(ndims + 1) = num_diurnal_samples
+      ndims = ndims + 1 !< Add one more dimension for the diurnal axis
+    endif
+
+    ! Allocate diag_buffer_obj, if it is not allocated.
```
Comment (uramirez8707) [inline diag_manager/fms_diag_object.F90:893]: Change comment to something like:
"Allocates diag_buffer_obj to the correct outputBuffer type based on the dimension (i.e outputBuffer0d_type, outputBuffer1d_type, etc ...) "

File: diag_manager/fms_diag_object.F90 [hunk 3]
```diff
@@ -849,4 +850,85 @@ subroutine dump_diag_obj( filename )
...
+    buffer_id = this%FMS_diag_fields(field_id)%buffer_ids(i)
+    ptr_diag_field_yaml => diag_yaml%get_diag_field_from_id(buffer_id)
+    num_diurnal_samples = ptr_diag_field_yaml%get_n_diurnal() !< Get number of diurnal samples
+
+    ! If diurnal axis exists, fill lengths of axes.
+    if (num_diurnal_samples .ne. 0) then
+      allocate(axes_length(ndims + 1)) !< Include extra length for the diurnal axis
+      do j = 1, ndims
+        axes_length(j) = this%fms_get_axis_length(axis_ids(j))
+      enddo
```
Comment (uramirez8707) [inline diag_manager/fms_diag_object.F90:888]: Add  "!TODO This is going to require more work for when we have subRegion variables "

We will figure this out later ...

File: diag_manager/fms_diag_object.F90 [hunk 4]
```diff
@@ -849,4 +850,85 @@ subroutine dump_diag_obj( filename )
...
+    if (.not. allocated(this%FMS_diag_output_buffers(buffer_id)%diag_buffer_obj)) then
+      this%FMS_diag_output_buffers(buffer_id) = fms_diag_output_buffer_create_container(ndims)
+    end if
+
+    ptr_diag_buffer_obj => this%FMS_diag_output_buffers(buffer_id)%diag_buffer_obj
+
+    select type (ptr_diag_buffer_obj)
+      type is (outputBuffer0d_type) !< Scalar buffer
+        if (allocated(ptr_diag_buffer_obj%buffer)) cycle !< If allocated, loop back
+        call ptr_diag_buffer_obj%allocate_buffer(field_data(1, 1, 1, 1), & !< If scalar field variable
```
Comment (uramirez8707) [inline diag_manager/fms_diag_object.F90:903]: @rem1776 can we initialize the buffer to the `FillValue`?

Comment (rem1776) [reply diag_manager/fms_diag_object.F90:906]: Yeah should be able to use `initialize_buffer` right after its allocated

Comment (ganganoaa) [reply diag_manager/fms_diag_object.F90:906]: Yes. I will update it soon.

**Pull request closed without merging**